examples/bayesian_regression.py

```diff
@@ -73,10 +73,12 @@ def model(data):
 
 
 def guide(data):
-    w_loc = torch.randn(1, p, requires_grad=True).type_as(data)
+    w_loc = torch.randn(1, p).type_as(data)
+    w_loc.requires_grad = True
```

Comment:
i dont think we need `requires_grad` because it is wrapped with `pyro.param`

Reply:
That'll make it cleaner. Great, I didn't realize the param store would handle that.